lucene/core/src/java/org/apache/lucene/search/Query.java

```diff
@@ -45,6 +46,13 @@
  */
 public abstract class Query {
 
+  static final VirtualMethod<Query> oldMethod =
+      new VirtualMethod<>(Query.class, "rewrite", IndexReader.class);
+  static final VirtualMethod<Query> newMethod =
+      new VirtualMethod<>(Query.class, "rewrite", IndexSearcher.class);
+  final boolean isDeprecatedMethodOverridden =
```

Comment:
Nit: make the variable name more specific?

```suggestion
  final boolean isDeprecatedRewriteMethodOverridden =
```